modules/adlooxRtdProvider.md

```diff
@@ -0,0 +1,88 @@
+# Overview
+
+    Module Name: Adloox RTD Provider
+    Module Type: RTD Provider
+    Maintainer: technique@adloox.com
+
+# Description
+
+RTD provider for adloox.com. Contact adops@adloox.com for information.
+
+In addition to populating the ad server key-value targeting, fetched segments (prefixed with `adl_...`) will also populate [First Party Data](https://docs.prebid.org/features/firstPartyData.html), some examples of the segments as described by the Adloox 'Google Publisher Tag Targeting Guidelines' and where they are placed are:
+
+ * Page/Device segments are placed into `fpd.context.data`, for example:
+     * **`adl_ivt`:** `fpd.context.data.adl_ivt` is a boolean
+     * **`adl_ua_old`:** `fpd.context.data.ua_old` is a boolean
+     * **`adl_ip`:** `fpd.context.data.adl_ip` is a string
+ * AdUnit segments are placed into `AdUnit.fpd.context.data`, for example:
+     * **`adl_{dis,vid,aud}`:** `AdUnit.fpd.context.data.adl_{dis,vid,aud}` is an array of integers
+     * **`adl_atf`:** `AdUnit.fpd.context.data.adl_atf` is a boolean (or `-1` on no measure)
+
+**N.B.** this provider does not offer or utilise any user orientated data
+
+This module adds an HTML `<script>` tag to the page to fetch our JavaScript from `https://p.adlooxtracking.com/gpt/a.js` (~3kiB gzipped) to support his integration.
+
+## Example
+
+To view an example of an Adloox integration look at the example provided in the [Adloox Analytics Adapter documentation](./adlooxAnalyticsAdapter.md#example).
+
+# Integration
+
+To use this, you *must* also integrate the [Adloox Analytics Adapter](./adlooxAnalyticsAdapter.md) as shown below:
+
+    pbjs.setConfig({
+      ...
+
+      realTimeData: {
+        auctionDelay: 700,             // recommended to be at least 700ms
```

Comment:
I do not like this recommendation at all. Nor the requirement that waitForIt to be true. Will discuss with the community.

The intention for RTD modules is that they are fast. 50ms. 100ms maybe. 700 is so far out of range it is likely to destroy a publishers ability to get successful bids.

Reply:
The 700ms is not for us, as described in the documentation below that, but as I have not seen publishers differentiate between office leased lines and contended mobile dongles we had to pick something; those that do use smarts would ignore this advice anyway. I settled on this number as there are only really two RTD modules with recommendations to glean from; `browsiRtdProvider.js` (1000ms) and `jwplayerRtdProvider.js` (100ms).

My understanding is that `auctionDelay` is only a deadline, if the bidders (and RTD sources) all return before that time limit then that phase is over and we move on, right?

We definitely do not intend to delay anything by 700ms and we would be just as comfortable with a ~100ms which is enough to cross either the US or the EU...my packet-pushing-hat does raise an eyebrow that a 3G/4G phone could reliably meet that deadline whilst simultaneously pulling in publisher content. Does it? Anyway, more than happy to lower the threshold warning or better still replace it with a warning that we missed the auction instead?

On the `waitForIt` note, can you describe the concern as the publisher controls this and not us. I do not understand how an RTD could function if the publisher is not willing to wait at all for it to call out to an API and pull in the segments to use during the auction process? I am missing something here, can you explain what I should be doing instead?

Maybe I am confused should be using one of and not both of `waitForIt` or `auctionDelay` and what I am doing wrong? I understood `auctionDelay` is a NOOP unless you have `waitForIt`?

The behaviour I am looking for is that we want a maximum of Xms (`auctionDelay`?) and if we exceed that our RTD does not get to participate.